plugins/tiddlywiki/freelinks/text.js

```diff
@@ -40,99 +55,210 @@ Compute the internal state of the widget
+		this.getVariable("tv-freelinks",{defaultValue:"no"}) === "yes" && 
+		!this.isWithinButtonOrLink()) {
+		
+		// Get the current tiddler title
+		var currentTiddlerTitle = this.getVariable("currentTiddler") || "";
+		
+		// Check if persistent caching is enabled
+		var persistCache = self.wiki.getTiddlerText(PERSIST_CACHE_TIDDLER, "no") === "yes";
+		var cacheKey = "tiddler-title-info-" + (ignoreCase ? "insensitive" : "sensitive");
+		
+		// Get the information about the current tiddler titles
+		this.tiddlerTitleInfo = persistCache ?
+			this.wiki.getPersistentCache(cacheKey, function() {
+				return computeTiddlerTitleInfo(self, ignoreCase);
+			}) :
+			this.wiki.getGlobalCache(cacheKey, function() {
+				return computeTiddlerTitleInfo(self, ignoreCase);
 			});
-			var regexpStr = "\\b(?:" + reparts.join("|") + ")\\b";
-			return {
-				titles: titles,
-				regexp: new RegExp(regexpStr,ignoreCase ? "i" : "")
-			};
-		});
-		// Repeatedly linkify
+		
+		// Process titles to avoid overlapping matches
 		if(this.tiddlerTitleInfo.titles.length > 0) {
-			var index,text,match,matchEnd;
-			do {
-				index = childParseTree.length - 1;
-				text = childParseTree[index].text;
-				match = this.tiddlerTitleInfo.regexp.exec(text);
-				if(match) {
-					// Make a text node for any text before the match
-					if(match.index > 0) {
-						childParseTree[index].text = text.substring(0,match.index);
-						index += 1;
+			var text = childParseTree[0].text,
+				newParseTree = [],
+				currentPos = 0;
+			
+			// Use Aho-Corasick to find all matches
+			var searchText = ignoreCase ? text.toLowerCase() : text;
+			var matches;
+			try {
+				matches = this.tiddlerTitleInfo.ac.search(searchText);
+			} catch(e) {
+				matches = [];
+			}
+			
+			// Sort matches by position and length (longer titles first)
+			matches.sort(function(a, b) {
+				if(a.index !== b.index) {
+					return a.index - b.index;
+				}
+				return b.length - a.length;
+			});
+			
+			// Process matches to avoid overlaps
+			var processedPositions = new Set();
+			for(var i = 0; i < matches.length; i++) {
+				var match = matches[i];
+				var matchStart = match.index;
+				var matchEnd = matchStart + match.length;
+				
+				// Skip if position already processed (ensures longer titles take precedence)
+				var overlap = false;
+				for(var pos = matchStart; pos < matchEnd; pos++) {
+					if(processedPositions.has(pos)) {
+						overlap = true;
+						break;
 					}
-					// Make a link node for the match
-					childParseTree[index] = {
+				}
+				if(overlap) {
+					continue;
+				}
+				
+				// Mark positions as processed
+				for(var pos = matchStart; pos < matchEnd; pos++) {
+					processedPositions.add(pos);
+				}
+				
+				// Add text before the match
+				if(matchStart > currentPos) {
+					newParseTree.push({
+						type: "plain-text",
+						text: text.slice(currentPos, matchStart)
+					});
+				}
+				
+				// Get matched title
```

Comment:
Let AI delete these comments, I use the prompt: Do not use repetitive comments that only describe the code content, only comments code that are difficult to discern the abstract design intent